cmake/HPX_SetupHIP.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c)      2020 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# No need for the user to specify the option explicitly
+if("${CMAKE_CXX_COMPILER}" MATCHES "hipcc$")
```

Comment:
And I see now that you actually already have it in `HPXConfig.cmake` but I think we should still only conditionally do some of the things in this file. Definitions should not need to be set up again (should have no effect since they get written to a header file) and the CMake options should already be there as cache variables.